### PR TITLE
Add additional overloads for mtimes (with handling for self-adjoint)

### PR DESCRIPTION
**Pull request by jmh530**
I'm open to different names for the enum that controls what to assume for the properties of the matrix.

### CODE REVIEW

**Comment by jmh530** [comment]
I moved what I did with dot into mtimes so that it would be similar to the original lubeck API.

There is some stuff in those original lubeck mtimes functions for handling the case where the kind isn't Contiguous that isn't getting handled here. It doesn't look like it's being tested though.

**Comment by jmh530** [comment]
@9il I think this is in good shape now if you have time to review.

**Comment by 9il** [comment]
Could it be new function names instead then extending API for the existing one?

**Comment by jmh530** [comment]
@9il I can make that change.

This PR also includes the addition of functionality from the original lubeck API to handle matrix-vector and vector-matrix and vector-vector multiplication. For the general matrix case, can I keep that as part of this PR? (i.e. just splitting off the selfAdjoint stuff to separate functions).

File: source/kaleidic/lubeck2.d [hunk 1]
```diff
@@ -71,87 +71,411 @@ unittest
...
 }
 
+/// ditto
 @safe pure nothrow @nogc
-Slice!(RCI!(Unqual!A), 2) mtimes(A, B, SliceKind kindA, SliceKind kindB)(
-    Slice!(const(A)*, 2, kindA) a,
-    auto ref const Slice!(RCI!B, 2, kindB) b
+Unqual!A mtimes(A, B, SliceKind kindA, SliceKind kindB)(
+    Slice!(const(A)*, 1, kindA) a,
+    auto ref const Slice!(RCI!B, 1, kindB) b
```
Comment (9il) [inline source/kaleidic/lubeck2.d:459]: GitHub doesn't show a proper diff. Please rearrange the new API at the end of the mtimes, block for a simpler review.

**Comment by jmh530** [comment]
@9il No rush, but when you have a chance can you push a new tag. I don't think I have permissions to do that on this repository.

**Pull request merged**